stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -1003,8 +1061,60 @@ public class MessageListViewModel(
          * @param user The user to be blocked.
          * @param cid The full channel id, i.e. "messaging:123".
          */
+        @Deprecated(
+            "Deprecated in order to bring make the action more explicit." +
+                "Use `MessageListViewModel.ShadowBanUser` if you want to keep the same functionality " +
+                "(shadow banning) or `MessageListViewModel.BanUser` if you want to ban a user.",
+            replaceWith = ReplaceWith(
+                "ShadowBanUser",
+                "package com.getstream.sdk.chat.viewmodel.messages.MessageListViewModel.ShadowBanUser"
+            )
+        )
         public data class BlockUser(val user: User, val cid: String) : Event()
 
+        /**
+         * When the user bans another user.
+         *
+         * @param user The user to be blocked.
+         * @param reason The reason for banning the user.
+         * @param timeout The time until the user is automatically unbanned, expressed in minutes.
+         * A null value represent an indefinite ban.
+         */
+        public data class BanUser(
```

Comment:
Unlike `Event.BlockUser`, I opet not to ask for CID here.
We have it provided inside `MessageListViewModel` and I think it maybe awkward to expect this VM to be able to block users belonging to a different channel.

I'm not sure why we conceived it differently initially, if you see me missing something here please comment. :smile: